pages/resources/tutorials/query-history.md

```diff
@@ -0,0 +1,189 @@
+# Retrieving historical block headers using Trin
+
+The Portal Network aims to provide a decentralized store of historical Ethereum data. It does this by distributing data across the network. Your Portal Network client provides you with a window into the network and request specific data from other nodes. In this guide you will learn how to start a Portal Network client and request some historical data. You will:
+
+- Install and run Trin
+- Learn how to retrieve data from your node
+- Learn how to decode Portal Network data
+
+## Prerequisites
+
+To retrieve data from the Portal network, you need to run a Portal client. In this tutorial you will use Trin. Trin is written in Rust, so your computer will need to have Rust installed.
+
+There are several options for how you send requests to your node. In this guide you will learn how to use HTTP to send requests. The requests are formatted as JSON data, and you will send them to your node using a tool called `curl`.
+
+The responses arrive at your node encoded, so you also have to decode them to make the data human-readable. You can use many different tools for this, in this tutorial we will use a simple Rust script. 
+
+To follow this guide you need to have the following installed on your machine:
+
+- [Rust](https://www.rust-lang.org/tools/install)
+- [cURL](https://curl.se/)
+- [Trin](https://github.com/ethereum/trin)
+
+There are specific instructions for installing Trin in the next section.
+
+
+## Install Trin
```

Comment:
This seems like maybe it should be placed in the `trin` repository and maintained there.  Seems like for tutorials like this to age well, we might avoid embedding this type of information in them.  Seems like a simple copy/paste fix to move this over to the trin repo (or to link to whatever tutorial exists over there already if there is one).